src/libstd/net/tcp.rs

```diff
@@ -323,6 +350,19 @@ impl TcpListener {
     /// This function will block the calling thread until a new TCP connection
     /// is established. When established, the corresponding `TcpStream` and the
     /// remote peer's address will be returned.
+    ///
+    /// # Examples
+    ///
+    /// ```no_run
+    /// use std::net::TcpListener;
+    ///
+    /// let listener = TcpListener::bind("127.0.0.1:8080").unwrap();
+    /// if let Ok((_socket, addr)) = listener.accept() {
+    ///     println!("new client: {:?}", addr);
+    /// } else {
+    ///     println!("cannot get new client :'(");
```

Comment:
After looking at the source, it looks like the `Err` case happens if `setsockopt` returns non-zero. I guess it's not that straightforward here.

On a side note, it took me a minute to realize this was a smiley face in this message. It might be more realistic if the example here was something like:

```rust
let listener = TcpListener::bind("127.0.0.1:8080").unwrap();
match listener.accept() {
    Ok((_socket, addr)) =>
        println!("new client: {:?}", addr)
    Err(e) =>
        println!("could not get client: {:?}", e)
}
```

Up to you though